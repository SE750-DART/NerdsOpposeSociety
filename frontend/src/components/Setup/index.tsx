import React, { CSSProperties } from "react";
import styles from "./style.module.css";

interface SetupProps {
  setupText: string;
}

const Setup = ({ setupText }: SetupProps) => {
  let setupSize: CSSProperties = {};
  if (setupText.length > 75) {
    setupSize = { fontSize: "24px" };
  } else if (setupText.length > 45) {
    setupSize = { fontSize: "32px" };
  }

  return (
    <div className={`${styles.setup}`}>
      <h5>The Setup:</h5>
      <h2 style={setupSize}>{setupText}</h2>
    </div>
  );
};

export default Setup;
