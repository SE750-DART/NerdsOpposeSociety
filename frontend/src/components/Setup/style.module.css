.setup {
  background: #bbdefb;
  background: linear-gradient(
    180deg,
    #bbdefb 0%,
    #bbdefb 80%,
    rgba(255, 255, 255, 0) 100%
  );
  max-width: 690px; /* magic number to limit width on desktop */
  padding: 24px 24px 36px 24px;
  position: sticky;
  top: 115px;
  transform: translateX(-24px);
  width: 100vw;
  z-index: 5;
}

.setupNoGradient {
  background: #bbdefb;
  padding-bottom: 24px;
}
