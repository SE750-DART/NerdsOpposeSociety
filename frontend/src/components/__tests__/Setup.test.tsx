import React from "react";
import renderer from "react-test-renderer";
import Setup from "../Setup";

it("Setup snapshot testing", () => {
  const shortSetup = "Daddy, why is mommy crying?";
  const longSetup =
    "After four platinum albums and three Grammys, it's time to get back to my roots, to what inspired me to make music in the first place: __________.";

  const treeShort = renderer.create(<Setup setupText={shortSetup} />).toJSON();
  expect(treeShort).toMatchSnapshot();

  const treeLong = renderer.create(<Setup setupText={longSetup} />).toJSON();
  expect(treeLong).toMatchSnapshot();
});
