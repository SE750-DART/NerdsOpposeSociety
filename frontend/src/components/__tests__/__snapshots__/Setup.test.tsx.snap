// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup snapshot testing 1`] = `
<div
  className="setup"
>
  <h5>
    The Setup:
  </h5>
  <h2
    style={Object {}}
  >
    Daddy, why is mommy crying?
  </h2>
</div>
`;

exports[`Setup snapshot testing 2`] = `
<div
  className="setup"
>
  <h5>
    The Setup:
  </h5>
  <h2
    style={
      Object {
        "fontSize": "24px",
      }
    }
  >
    After four platinum albums and three Grammys, it's time to get back to my roots, to what inspired me to make music in the first place: __________.
  </h2>
</div>
`;
